pkg/controllers/dynakube/dynakube_controller.go

```diff
@@ -329,23 +352,29 @@ func (controller *Controller) reconcileConnectionInfo(ctx context.Context, dynak
 }
 
 func (controller *Controller) reconcileComponents(ctx context.Context, dynatraceClient dtclient.Client, dynakube *dynatracev1beta1.DynaKube) error {
-	err := controller.reconcileActiveGate(ctx, dynakube, dynatraceClient)
+	// it's important to setup app injection before AG so that it is already working when AG pods start, in case code modules shall get
```

Comment:
order change makes sense for now, but I would like to repeat my earlier question:
> would this kind of logic work in case we had different CR for activegate?

(considering "different CR" == "different Controller")

Reply:
There is a dependency between AG and App monitoring in case AG shall be monitored with code modules. This has to be taken into account, when we split the controller up. At the moment I don't have an idea how to solve this, because when deploying AG one would have to check that app monitoring is operational and that connection infos are available at whatever location they are placed by then. And one would need to know if AG should be monitored with code modules in the first place (aka. read the related feature flag of a different CR).
TL;DR; it looks like that's gonna be a challenge, but one that's for sure out of scope in this PR.